src/Compilers/VisualBasic/Portable/Operations/VisualBasicOperationFactory_Methods.vb

```diff
@@ -439,6 +439,28 @@ Namespace Microsoft.CodeAnalysis.Semantics
                 [event], instance, Create(statement.Handler), adds:=False, isInvalid:=statement.HasErrors, syntax:=statement.Syntax, type:=Nothing, constantValue:=Nothing)
         End Function
 
+        Private Function GetConversionKind(kind As VisualBasic.ConversionKind) As Semantics.ConversionKind
+            Dim operationKind = Semantics.ConversionKind.Invalid
+
+            If kind.HasFlag(VisualBasic.ConversionKind.UserDefined) Then
+                operationKind = Semantics.ConversionKind.OperatorMethod
+            ElseIf Conversions.IsIdentityConversion(kind) Or
+                   kind.HasFlag(VisualBasic.ConversionKind.Reference) Or
+                   kind.HasFlag(VisualBasic.ConversionKind.TypeParameter) Or
+                   kind.HasFlag(VisualBasic.ConversionKind.Array) Or
+                   kind.HasFlag(VisualBasic.ConversionKind.Value) Then
+                operationKind = Semantics.ConversionKind.Cast
```

Comment:
Can the condition be written as `kind.HasFlag(ConversionKind.Reference Or ConversionKind.TypeParameter Or ConversionKind.Array Or ConversionKind.Value)`?

If not, use `OrElse` rather than `Or`.